parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -33,23 +92,128 @@ decl_event! {
 
 decl_error! {
 	pub enum Error for Module<T: Config> {
+		/// The message payload exceeds byte limit.
+		PayloadTooLarge,
+		/// No more messages can be queued for the channel during this commit cycle.
+		QueueSizeLimitReached,
+		/// Cannot increment nonce
+		Overflow,
+		/// Not authorized to send message
+		NotAuthorized,
 	}
 }
 
 decl_module! {
 	pub struct Module<T: Config> for enum Call where origin: T::Origin {
 		type Error = Error<T>;
 		fn deposit_event() = default;
+
+		// Generate a message commitment every [`Interval`] blocks.
+		//
+		// The commitment hash is included in an [`AuxiliaryDigestItem`] in the block header,
+		// with the corresponding commitment is persisted offchain.
+		fn on_initialize(now: T::BlockNumber) -> Weight {
+			if (now % Self::interval()).is_zero() {
+				Self::commit()
+			} else {
+				T::WeightInfo::on_initialize_non_interval()
+			}
+		}
+
+		#[weight = T::WeightInfo::set_principal()]
+		pub fn set_principal(origin, principal: <T::Lookup as StaticLookup>::Source) -> DispatchResult {
+			T::SetPrincipalOrigin::ensure_origin(origin)?;
```

Comment:
does this check enforce that the original is the current principal?

Reply:
Not really possible technically. The authorised origin (root or council) can never be the principal.

However the principal can be a member of the council.

Reply:
what does this line/check do?

Reply:
ensures that the dispatch origin is either root or a quorum of council members.

Reply:
@vgeddes I guess we could also allow the principal to change the principal – would that make sense? 

Regarding implementation, did you see this: https://github.com/paritytech/substrate/blob/master/frame/assets/src/lib.rs#L479-L482

Reply:
I'd rather keep things simple for now and only authorize government agents (root or council). In practice, the principal is likely going to be on the council anyway, where they can propose a new principal or influence the vote.